# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsURLHelper.h
# Commit: c2ca2f596c09
# Full Hash: c2ca2f596c0922e7fd34494dc6ed8dd68d46b122
# Author: abhishekmadan30 <148475204+abhishekmadan30@users.noreply.github.com>
# Date: 2024-03-26 04:11:24
# Regressor Bug: 1877419
# File Overlap Count: 1
# Description:
#   Bug 1877419 - Part 1: Added nested stripping of urls to copy without site tracking. r=pbz,necko-reviewers
#   
#   Depends on D204914
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202943
# ==============================================================================

diff -r 45931e6db21e -r c2ca2f596c09 netwerk/base/nsURLHelper.h
--- a/netwerk/base/nsURLHelper.h	Mon Mar 25 18:30:38 2024 +0000
+++ b/netwerk/base/nsURLHelper.h	Mon Mar 25 18:30:38 2024 +0000
@@ -308,6 +308,8 @@
    */
   void Serialize(nsAString& aValue, bool aEncode) const;
 
+  static void SerializeString(const nsCString& aInput, nsAString& aValue);
+
   void Get(const nsAString& aName, nsString& aRetval);
 
   void GetAll(const nsAString& aName, nsTArray<nsString>& aRetval);
@@ -338,6 +340,8 @@
 
   uint32_t Length() const { return mParams.Length(); }
 
+  static void DecodeString(const nsACString& aInput, nsAString& aOutput);
+
   const nsAString& GetKeyAtIndex(uint32_t aIndex) const {
     MOZ_ASSERT(aIndex < mParams.Length());
     return mParams[aIndex].mKey;
@@ -355,7 +359,6 @@
   void Sort();
 
  private:
-  static void DecodeString(const nsACString& aInput, nsAString& aOutput);
   static void ConvertString(const nsACString& aInput, nsAString& aOutput);
   static bool ParseNextInternal(const char*& aStart, const char* aEnd,
                                 bool aShouldDecode, nsAString* aOutputName,